src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+
+        var vmrProductRepo = Path.Combine(tmpPath, Guid.NewGuid().ToString());
+        if (vmr.IsLocal)
+        {
+            await CheckoutBranch(vmr);
+            CopyDirectory(Path.Combine(vmr.Path, VmrInfo.SourceDirName, mapping), vmrProductRepo, true);
+        }
+        else
+        {
+            vmrProductRepo = await PartiallyCloneVmr(tmpPath, vmr, mapping);
+        }
+        await GitInitRepo(vmrProductRepo, vmr.Branch);
+
+        return vmrProductRepo;
+    }
+
+    private async Task<string> PrepareRepo(DiffRepo repo, string tmpPath)
+    {
+        string tmpProductRepo = Path.Combine(tmpPath, Path.GetFileName(repo.Path));
+
+        if (!repo.IsLocal)
+        {
+            await processManager.ExecuteGit(Path.GetDirectoryName(tmpProductRepo)!, [
+                "clone",
+                "--depth", "1",
+                repo.Path,
+                "-b", repo.Branch,
+                tmpProductRepo
+            ]);
+        }
+        else
+        {
+            await CheckoutBranch(repo);
+            CopyDirectory(repo.Path, tmpProductRepo, true);
+        }
+
+        return tmpProductRepo;
+    }
+
+    private async Task<(DiffRepo repo1, DiffRepo repo2)> ParseInput()
+    {
+        DiffRepo repo1, repo2;
+        var parts = options.Input.Split("..", StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length > 2 || parts.Length < 1)
+        {
+            throw new ArgumentException($"Invalid input {options.Input}");
+        }
+        
+        if (parts.Length == 1)
+        {
+            var currentPath = Directory.GetCurrentDirectory();
+            var res = await processManager.ExecuteGit(currentPath, "status");
+            res.ThrowIfFailed("Current directory is not a git repo");
+            repo1 = new DiffRepo(currentPath, string.Empty, true, IsLocalRepoVmr(currentPath));
+            repo2 = await ParseRepo(parts[0]);
+        }
+        else
+        {
+            repo1 = await ParseRepo(parts[0]);
+            repo2 = await ParseRepo(parts[1]);
+        }
+        await VerifyInput(repo1, repo2);
+
+        return (repo1, repo2);
+    }
+
+    private async Task<IReadOnlyCollection<string>> GetDiffFilters(string mapping)
+    {
+        var gitRepo = gitRepoFactory.CreateClient(DarcLib.Constants.DefaultVmrUri);
+        return sourceMappingParser.ParseMappingsFromJson(
+            (await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", DarcLib.Constants.DefaultVmrUri, "main")))
+            .Where(m => m.Name == mapping)
+            .First().Exclude;
+    }
+
+    private async Task CheckoutBranch(DiffRepo repo)
+    {
+        if (!string.IsNullOrEmpty(repo.Branch))
+        {
+            var res = await processManager.ExecuteGit(repo.Path, [
+                    "checkout",
+                    repo.Branch
+                ]);
+            res.ThrowIfFailed($"Failed to checkout requested branch {repo.Branch} in {repo.Path}");
+        }
+    }
```

Comment:
As a rule, it's weird to have the `CheckoutBranch` not do anything for no branch. It's better if the checkout method just always checks out. We can if/else around it from wherever we use it if we think it's possible we don't want to check out anything.

Also we have a method for this on `ILocalGitRepo` so we should just use that.